pyomo/dae/flatten.py

```diff
@@ -123,15 +123,18 @@ def flatten_dae_variables(model, time):
             for _slice in generate_time_indexed_block_slices(b, time):
                 time_indexed_vars.append(Reference(_slice))
             continue
-        block_queue.extend(
-            list(b.component_objects(Block, descend_into=False))
-        )
-        for v in b.component_objects(SubclassOf(Var), descend_into=False):
-            v_sets = v.index_set().subsets()
-            if time in v_sets:
-                for _slice in generate_time_only_slices(v, time):
-                    time_indexed_vars.append(Reference(_slice))
-            else:
-                regular_vars.extend(list(v.values()))
+        for blkdata in b.values():
+            block_queue.extend(
+                list(blkdata.component_objects(Block, descend_into=False))
```

Comment:
random comment, but fwiw, you don't need the cast to `list` here. `extend()` will happily take an iterable.